modules/internal/localeModels/knl/LocaleModel.chpl

```diff
@@ -513,13 +513,13 @@ module LocaleModel {
     proc localeIDtoLocale(id : chpl_localeID_t) {
       const node = chpl_nodeFromLocaleID(id);
       const subloc = chpl_sublocFromLocaleID(id);
-      if (subloc == c_sublocid_none) || (subloc == c_sublocid_any) then
-        return (myLocales[node:int]):locale;
+      if chpl_isActualSublocID(subloc) then
+        return (myLocales[node:int].getChild(subloc:int)):locale;
       else if (subloc == numaDomainForAny(
                         (myLocales[node:int]:LocaleModel).numSublocales)) then
```

Comment:
Oops, yes, I missed that -- bizarrely, I was thinking that the magic `numaDomainForAny()` value was also not an "actual sublocale ID".  Working on a fix ...